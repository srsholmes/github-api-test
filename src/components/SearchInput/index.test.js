import React from 'react';
import {render, getByTestId, fireEvent} from '@testing-library/react';
import {SearchInput} from './index';

describe('SearchInput', () => {
  it('fires the function passed to by the setInput Prop', () => {
    const onChange = jest.fn();
    const {container} = render(
      <SearchInput value={'hello'} setInput={onChange} />
    );
    const input = getByTestId(container, 'search-input');
    expect(input.value).toBe('hello');
    fireEvent.change(input, {target: {value: 'testInput'}});
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
