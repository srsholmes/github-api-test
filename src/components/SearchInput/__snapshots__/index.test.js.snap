// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput displays only the loader 1`] = `
<div>
  <h3
    class="title"
  >
    Please search for a Github User:
  </h3>
  <input
    class="input"
    data-testid="search-input"
    type="text"
    value="test"
  />
</div>
`;
